Test and document usage of BP_NPM_START_SCRIPT

c0d1ngm0nk3y opened the pull request:
<!-- Thanks for contributing. To speed up the process of reviewing your pull
request please provide us with the following information: -->

## Summary
<!-- A short explanation of the proposed change -->
`BP_NPM_START_SCRIPT` was already being used, but hard to discover.

## Use Cases
<!-- An explanation of the use cases your change enables -->
No new functionality.

## Checklist
<!-- Please confirm the following -->
* [X] I have viewed, signed, and submitted the Contributor License Agreement.
* [ ] I have linked issue(s) that this PR should close using keywords or the Github UI (See [docs](https://docs.github.com/en/github/managing-your-work-on-github/linking-a-pull-request-to-an-issue))
* [X] I have added an integration test, if necessary.
* [X] I have reviewed the [styleguide](https://github.com/paketo-buildpacks/community/blob/main/STYLEGUIDE.md) for guidance on my code quality.
* [X] I'm happy with the commit history on this PR (I have rebased/squashed as needed).

c0d1ngm0nk3y commented:
@paketo-buildpacks/nodejs-maintainers Could you have a look? No functional change, only a small change to make the feature more discoverable.

thitch97 commented on README.md:
```diff
@@ -56,6 +56,15 @@ the `BP_NODE_PROJECT_PATH` environment variable at build time either directly
 
+## Specifying a custom start script
+
+To specify a project subdirectory to be used as the root of the app, please use
```
I don't think this section matches its title, seems copied from above.

c0d1ngm0nk3y replied:
> I don't think this section matches its title, seems copied from above.

True. There was something missing. I updated the PR. Thanks.

thitch97 commented on integration/testdata/app_with_start_cmd/package.json:
```diff
@@ -16,7 +16,8 @@
     "poststart": "echo \"poststart\"",
     "prestart": "echo \"prestart\"",
-    "start": "echo \"start\" && node server.js",
+    "start": "echo \"start\" && node hello`.js",
```
@c0d1ngm0nk3y Looks like there's an extra backtick here that's causing the tests to fail

thitch97 commented:
Happy to merge this once the integration tests are passing!

c0d1ngm0nk3y commented:
> Happy to merge this once the integration tests are passing!

Sorry for the delay. Was on vacation. I have fixed the testdata and rebased the pr. Locally, the tests are fine now.

c0d1ngm0nk3y commented:
> Happy to merge this once the integration tests are passing!

All tests seem to pass now :)

c0d1ngm0nk3y commented:
@paketo-buildpacks/nodejs-maintainers Any chance of getting this merged? All tests look fine and it is basically only missing documentation/test of an existing feature.
